integration/docker/export_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docker
+
+import (
+	"fmt"
+	. "github.com/clearcontainers/tests"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"os"
+	"os/exec"
+)
+
+var _ = Describe("export", func() {
+	var (
+		args     []string
+		id       string
+		exitCode int
+		command  *Command
+	)
+
+	runCommand := func(args []string, expectedExitCode int) {
+		command = NewCommand(Docker, args...)
+		Expect(command).ToNot(BeNil())
+		exitCode = command.Run()
+		Expect(exitCode).To(Equal(expectedExitCode))
+	}
+
+	BeforeEach(func() {
+		id = RandID(30)
+		args = []string{"run", "-td", "--name", id, Image}
+		runCommand(args, 0)
+	})
+
+	AfterEach(func() {
+		Expect(ContainerRemove(id)).To(BeTrue())
+		Expect(ContainerExists(id)).NotTo(BeTrue())
+	})
+
+	Describe("export with docker", func() {
+		Context("export a container", func() {
+			It("should export filesystem as a tar archive", func() {
+				args = []string{"export", "--output=/tmp/latest.tar", id}
+				runCommand(args, 0)
+				_, err := os.Stat("/tmp/latest.tar")
+				if err != nil {
+					fmt.Printf("ERROR: %v\n", err)
+					os.Exit(1)
+				}
+				cmd := exec.Command("rm", "-rf", "/tmp/latest.tar")
```

Comment:
of course, it would be great if we could somehow check if the contents of the tar matched the container we ran up. Maybe we can do an exec into the container to generate a specifically named file, and then check that that file ends up in the tar file?